cmd/disk-cache.go

```diff
@@ -0,0 +1,764 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"time"
+
+	errors2 "github.com/minio/minio/pkg/errors"
+	"github.com/minio/minio/pkg/wildcard"
+
+	"github.com/minio/minio/pkg/hash"
+)
+
+// list of all errors that can be ignored in tree walk operation in disk cache
+var cacheTreeWalkIgnoredErrs = append(baseIgnoredErrs, errDiskAccessDenied, errVolumeNotFound, errFileNotFound)
+
+const (
+	// disk cache needs to have cacheSizeMultiplier * object size space free for a cache entry to be created.
+	cacheSizeMultiplier      = 100
+	maxDiskUsagePct          = 80 // in %
```

Comment:
you can call this `cacheMaxDiskUsage` so that all cache global vars start with `cache`